src/main/java/org/opentripplanner/updater/spi/TimetableSnapshotFlush.java

```diff
@@ -0,0 +1,37 @@
+package org.opentripplanner.updater.spi;
+
+import org.opentripplanner.ext.siri.SiriTimetableSnapshotSource;
+import org.opentripplanner.updater.trip.TimetableSnapshotSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Flush the timetable snapshot buffer by committing pending changes.
+ */
+public class TimetableSnapshotFlush implements Runnable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TimetableSnapshotFlush.class);
+
+  private final SiriTimetableSnapshotSource siriTimetableSnapshotSource;
+  private final TimetableSnapshotSource gtfsTimetableSnapshotSource;
+
+  public TimetableSnapshotFlush(
+    SiriTimetableSnapshotSource siriTimetableSnapshotSource,
+    TimetableSnapshotSource gtfsTimetableSnapshotSource
+  ) {
+    this.siriTimetableSnapshotSource = siriTimetableSnapshotSource;
+    this.gtfsTimetableSnapshotSource = gtfsTimetableSnapshotSource;
+  }
+
+  @Override
+  public void run() {
+    LOG.debug("Flushing timetable snapshot buffer");
+    if (siriTimetableSnapshotSource != null) {
+      siriTimetableSnapshotSource.flushBuffer();
+    }
+    if (gtfsTimetableSnapshotSource != null) {
+      gtfsTimetableSnapshotSource.flushBuffer();
+    }
+    LOG.debug("Flushed timetable snapshot buffer");
```

Comment:
If this run method throws any exception, even due to a temporary problem, the executor service will stop running it at a fixed delay. I think the body of the run method should be wrapped in a try/catch block that logs and discards the exceptions. For example if somehow the indexing and commit operation accidentally causes a divide by zero error. Maybe even catch Throwable if there's any Error subclass we need to log and discard.

Reply:
You're right, wrapping the runnable in a try/catch prevents the scheduling from stopping abruptly.

In this context catching Exceptions rather than Throwables seems more appropriate to me. This can be adjusted later if needed.